user/mongoStoreClient.go

```diff
@@ -219,6 +221,47 @@ func (msc *MongoStoreClient) FindUsersByRole(role string) (results []*User, err
 	return results, nil
 }
 
+func (msc *MongoStoreClient) FindUsersByRoleAndDate(role string, createdFrom string, createdTo string) (results []*User, err error) {
+	opts := options.Find().SetCollation(usersCollation)
+
+	query := []bson.D{
+		bson.D{{Key: "roles", Value: role}},
+	}
+
+	if createdFrom != "" {
+		if timeFrom, err := time.Parse("2006-01-02", createdFrom); err != nil {
+			log.Println("Failed to parse time:", timeFrom)
+		} else {
+			fromObjectID := primitive.NewObjectIDFromTimestamp(timeFrom)
+			query = append(query, bson.D{{Key: "_id", Value: bson.M{"$gte": fromObjectID}}})
+		}
+	}
+	if createdTo != "" {
+		if timeTo, err := time.Parse("2006-01-02", createdTo); err != nil {
+			log.Println("Failed to parse time:", timeTo)
```

Comment:
I think it's better if you fail the request instead of silently ignoring invalid inputs